src/android/org/sunbird/support/SunbirdFileHandler.java

```diff
@@ -245,11 +242,10 @@ private static boolean isNullOrEmpty(String string) {
         return string == null || string.length() == 0;
     }
 
-    public static File getRequiredDirectory(File externalFilesDir, String directoryName) {
-        File directory = new File(externalFilesDir.getPath() + "/" + directoryName);
-
+    public static File getRequiredDirectory(Context context, String directoryName) {
+        File directory = new File(context.getExternalFilesDir(null).getAbsolutePath() + '/' + directoryName);
```

Comment:
This might break if context.getExternalFilesDir(null) is null so add a null check